[Backport 1.x] Opensearch dashboards ingress template fix to allow named port

peterzhuamazon opened the pull request:
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
[Backport 1.x] Opensearch dashboards ingress template fix to allow named port
 
### Issues Resolved
https://github.com/opensearch-project/helm-charts/pull/312
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

sastorsl commented:
Breaks deployment.

```
Helm upgrade failed: error validating "": error validating data: ValidationError(Ingress.spec.rules[0].http.paths[0].backend.se
rvice): unknown field "port:number" in io.k8s.api.networking.v1.IngressServiceBackend
```

#324 and a 1.x version of #325 should be added.

peterzhuamazon commented:
> 

Fixed now

> Breaks deployment.
> 
> ```
> Helm upgrade failed: error validating "": error validating data: ValidationError(Ingress.spec.rules[0].http.paths[0].backend.se
> rvice): unknown field "port:number" in io.k8s.api.networking.v1.IngressServiceBackend
> ```
> 
> #324 and a 1.x version of #325 should be added.

Fixed now.

sastorsl commented:
Can confirm that #327 fixed the issue.
Successfully deployed now.
